mobipick_gazebo/nodes/robotiq_2f_140_command_bridge.py

```diff
@@ -56,47 +56,73 @@ def set_jt_goal(self, position, effort=30.0):
     def execute_cb(self, goal):
         # helper variables
         rospy.loginfo(
-            "Recieved gripper command with position=%.3f and effort=%.3f"
+            "Received gripper command with position=%.3f and effort=%.3f"
             % (goal.command.position, goal.command.max_effort)
         )
         r = rospy.Rate(10)
         success = True
         self.set_jt_goal(-0.76 / 0.14 * goal.command.position + 0.76, goal.command.max_effort)
         self._jt_client.send_goal(self._actual_goal)
 
-        # start executing the action
-        while not self._jt_client.get_state() == actionlib.SimpleGoalState.ACTIVE:
-            # check that preempt has not been requested by the client
+        # Start executing the action
+        while not rospy.is_shutdown():
+            # Check if the action client has finished
+            state = self._jt_client.get_state()
+            if state in [
+                actionlib.GoalStatus.SUCCEEDED,
+                actionlib.GoalStatus.ABORTED,
+                actionlib.GoalStatus.PREEMPTED,
+                actionlib.GoalStatus.REJECTED,
+            ]:
+                break  # Stop if client reports the goal is already done
+
             if self._as.is_preempt_requested():
                 rospy.loginfo('%s: Preempted' % self._action_name)
-                self._as.set_preempted()
-                success = False
+                self._jt_client.cancel_goal()  # Cancel the goal on the action client
+                self._as.set_preempted()  # Set the action server to preempted
                 self._feedback.stalled = True
+                return  # Exit the execute_cb method
+
+            # Update feedback
             self._feedback.reached_goal = False
-            self._feedback.position = 0.14 - 0.14 / 0.76 * self._joint_state.actual.positions[0]
-            # publish the feedback
+            if hasattr(self, '_joint_state') and self._joint_state.actual.positions:
+                self._feedback.position = 0.14 - 0.14 / 0.76 * self._joint_state.actual.positions[0]
+            else:
+                self._feedback.position = 0.0  # Default value if joint state is unavailable
+
+            # Publish the feedback
             self._as.publish_feedback(self._feedback)
-            # this step is not necessary, the sequence is computed at 1 Hz for demonstration purposes
             r.sleep()
 
-        self._jt_client.wait_for_result()
```

Comment:
This introduces a bug! Don't delete this line.

Deleting the `wait_for_result` line will cause problems when we call `get_result` while no result has been received yet. Could also be related to the new issue (`Received comm state PREEMPTING when in simple state DONE with SimpleActionClient in NS /mobipick/gripper_controller/follow_joint_trajectory`)?

We definitely have to wait for the result here.